Add QFileSystem support in utils

nk0leg opened the pull request:
as part the task SDLWIN-133

Commit:
Added QFileSystem support in utils

as part the task SDLWIN-133

AGaliuzov commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+
+int64_t file_system::FileSize(const std::string &path) {
+  if (file_system::FileExists(path)) {
+    return static_cast<uint64_t>(QFileInfo(QString(path.c_str())).size());
```
why you cast `qint64` to `uint64_t` and then to `int64_t`?

AGaliuzov commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+     if (str_info.isDir()){
+         QDir dir(QString(path.c_str()));
+         QFileInfoList list = dir.entryInfoList(QDir::Files | QDir::Dirs |  QDir::Hidden | QDir::NoSymLinks | QDir::NoDotAndDotDot);
+         for (int i = 0; i < list.size(); ++i) {
```
can we use `Qts` `foreach` instead of raw loop?

malirod replied:
I don't like this. The lesser we use Qt the better.

AGaliuzov replied:
This is Qt port why not use good features from framework?

malirod replied:
Personally I don't like Qt in general. If we would use c++11 ever, then we will have native foreach... Lets vote ;)

AGaliuzov replied:
ok my hand is up.
And also Oleg uses `foreach` in this code below

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+#include <QUrl>
+
+#include <io.h>
+// TODO(VS): lint error: Streams are highly discouraged.
```
I think we can drop this todo

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+#define R_OK    4
+#define W_OK    2
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "Utils")
```
Propose to add extend logger name. Something like "Utils.FileSystem"

nk0leg replied:
if it necessary 
may be need make change for all components?

malirod replied:
Ok. lets leave as it is. Since win port is not the main repo. In the genivi such some changes regarding logger names were already done.

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+CREATE_LOGGERPTR_GLOBAL(logger_, "Utils")
+
+uint64_t file_system::GetAvailableDiskSpace(const std::string& path) {
+ QStorageInfo * mstor = new QStorageInfo(QString(path.c_str()));
```
move `*` to the type

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+uint64_t file_system::GetAvailableDiskSpace(const std::string& path) {
+ QStorageInfo * mstor = new QStorageInfo(QString(path.c_str()));
+ qint64 b_aval = mstor->bytesAvailable();
+ if (b_aval != 0) {
```
This if block is redundant. just return `static_cast<uint64_t>(b_aval);`

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+  }
+}
+
+int64_t file_system::FileSize(const std::string &path) {
```
move ref to the type

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+}
+
+size_t file_system::DirectorySize(const std::string& path) {
+    quint64 size = 0;
```
Fix indent

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+std::string file_system::CreateDirectory(const std::string& name) {
+  if (!DirectoryExists(name)) {
+     QDir dir;
+     if(dir.mkdir(QString(name.c_str()))){
```
add space before `{`

malirod replied:
Is it required to create QString explicitly?
**Update:** Here and in other places

nk0leg replied:
Yes. it must be.
Because constructor of QDir for QString & QDir only

AGaliuzov replied:
But I believe that Qtsring can be created from const char\* implicitly

malirod replied:
I had the same feeling. But this Qt (... these guys have a lot of surprises.

AGaliuzov replied:
It will diffenetly work, since I use same stuf for QSql

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+
+bool file_system::DirectoryExists(const std::string& name) {
+    QFileInfo checkFile(QString(name.c_str()));
+    if (checkFile.exists() && checkFile.isDir()) {
```
Fix indent in all places. Here and in other methods. Please double check all methods.
Replace if block with one liner.

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+
+bool file_system::FileExists(const std::string& name) {
+     QFileInfo checkFile(QString(name.c_str()));
+     if (checkFile.exists() && checkFile.isFile()) {
```
Replace with one liner

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+    return true;
+  }
+}
+
```
extra line

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+  if (!FileExists(src) || FileExists(dst) || !CreateFile(dst)) {
+    return false;
+  }
+    return QFile::copy(QString(src.c_str()), QString(dst.c_str()));
```
indent

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+}
+
+std::string file_system::GetPathDelimiter() {
+  return "\\";
```
Use qt native way. there is some thing like `QDir::delimiter`(not sure exact member)

nk0leg replied:
Exists QDir::separator(), but him returned QChar.
And for order to get std::string necessary something like : QDir::separator().decomposition().toStdStrig().
May be simply as there?

malirod replied:
We have to use Qt native and cross-platform solution. Possibly it worth to return just "\" on windows and update client code.

nk0leg replied:
see eb76be2

malirod commented on src/components/utils/src/file_system_qt.cc:
```diff
@@ -0,0 +1,357 @@
+}
+
+void file_system::MakeAbsolutePath(std::string& path){
+    if (!IsRelativePath(path)) {
```
indent

nk0leg replied:
see eb76be2

malirod commented:
@nk0leg, please, split commits in future In this PR there can be at least two. One for changes in cmake files, another for file system implementation.

Commit:
Add /EHsc key for WIN_NATIVE branch
Formating utils/CMakeFile
Correction file_system_qt.cc in accordance with style guide and remove redundancy code

Commit:
Fix formatting in accordance with style guide

Commit:
Correctation CMakeFile

Commit:
add space in line 76

anosach-luxoft commented:
@nk0leg src/components/utils/src/file_system_qt.cc:76 causes a build error
Line 339 too

Commit:
Fix build file_system_qt.cc
add 'const' in line 339
add '|' in line 76

nk0leg commented:
see 7465b17

---

This e-mail and any attachment(s) are intended only for the recipient(s) named above and others who have been specifically authorized to receive them. They may contain confidential information. If you are not the intended recipient, please do not read this email or its attachment(s). Furthermore, you are hereby notified that any dissemination, distribution or copying of this e-mail and any attachment(s) is strictly prohibited. If you have received this e-mail in error, please immediately notify the sender by replying to this e-mail and then delete this e-mail and any attachment(s) or copies thereof from your system. Thank you.
